Fix pre 1.7.1 error (purgeFull vs. purge-full)

vorlon opened the pull request:
This adds a duply_version variable which is filled with defaults for Ubuntu and Debian. It is then used to decide on the command in the cron job. This is hopefully not too simple. Has been tested on Debian wheezy and jessie. Should resolve #4

tohuwabohu commented on manifests/profile.pp:
```diff
@@ -257,9 +257,16 @@
     ensure  => present,
   }
 
+  if versioncmp($duplicity::params::duply_version, '1.7.1') < 0 {
```
The version is currently based on the operating system only. That is not enough. For instance I can easily install the latest version 1.9.1 on Debian 7 (via the archive) and yet the version check would think it is 1.5.5.5.

I suggest to add block like the following to the `init.pp` which does the distinction between the archive and the regular package and uses the right version

```
  $real_duply_version = empty($duply_version) ? {
    true => $duply_package_provider ? {
      archive => $duply_archive_version,
      default => $duplicity::params::duply_version,  # fall back to the default ...
    },
    default   => $duply_version,  # in case the default package version is wrong
  }
```

This would also allow to give the module a hint in case the platform is not supported. What do you think?

Other than that it looks pretty good!

tohuwabohu commented:
@vorlon ?

tohuwabohu commented:
@vorlon ?

vorlon commented:
@tohuwabohu sorry for the silence
You are right, some changes are needed here, I will get back to you about it

vorlon commented:
What about something like

    if ($duply_package_provider == 'archive') and ($duply_archive_version != $duplicity::params::duply_version) {
      $duply_version = $duply_archive_version
      }
      else {
        $duply_version = $duplicity::params::duply_version
      }

in init.pp and using $duplicity::duply_version in profile.pp?

tohuwabohu commented:
Looks good to me.

luxflux commented:
Any news here? I don't want to install it from source :(

vorlon commented:
Just added the duply_version parameter as @tohuwabohu suggested, I also included some tests for the cron job when using different duply versions. To make testing possible in an easy way I also added a duply_version and duply_executable parameter to profile.pp.

tohuwabohu commented:
@vorlon, can you please have a look at the merge conflict? thanks

tohuwabohu commented on manifests/profile.pp:
```diff
@@ -86,6 +92,8 @@
   $cron_enabled        = $duplicity::cron_enabled,
   $cron_hour           = undef,
   $cron_minute         = undef,
+  $duply_executable    = $duplicity::real_duply_executable,
```
I've removed the `real_duply_executable` on master as it was causing some trouble (#10) and made things unnecessarily complex. Instead, the `PATH` environment variable is set.

I suggest to change the default of the parameter to `duply`.

vorlon commented:
@tohuwabohu conflicts have been resolved and this works for me on Debian jessie. Please double check.

tohuwabohu commented:
I've released version 3.6.0 to the forge. Thanks for the contribution.
